lib/contracts.ts

```diff
@@ -93,7 +93,7 @@
~  const log = receipt.logs.find((log) =>
~    log.topics.every((val, i) => val === topics[i])
~  );
~  return contract.interface.decodeEventLog(fragment, log!.data);
~}
~
~export function getRequestEvent(
```

Comment:
Is the obligation the owner of its option contract? Looking at the code it seems that it would rather be the `OptionPairFactory` but I might have missed something.

Reply:
Correct, when creating a pair the Factory assigns the Obligation as the owner.

Reply:
All good then!